### DISCUSSION TITLE
Why is Iced so verbose?

### DISCUSSION THREAD

**Discussion by Kesanov** [hecrj/iced#1031] [Q&A] [answered]
I don't understand why Iced is so verbose:

- The constructors: `Button::new, Text::new, ..` instead of `button, text`.
- The builder patten: `column.push(a).push(b)` instead of Elmy `column([a,b])`.

It doesn't improve performance noticeably and makes code considerably harder to read.

**Comment by 90-008** [comment]
If you dislike it, you can have your own methods that create widgets like you said. `Column::with_children(vec![a, b])` exists for "elmy". I personally have macros and some functions for these, you can see some on [one of my apps](https://github.com/harmony-development/Crust/blob/master/src/component/mod.rs#L150)

**Comment by Kaiden42** [comment]
> The constructors: Button::new, Text::new, .. instead of button, text.

That's because Rust and Elm are two completely different languages. Iced is a GUI library for Rust, not Elm. Therefore it's using the Rust syntax. Yes, it is possible to implement macros like `button!(...)` or `text!(...)`, but using macros has some downsides. They don't have fully support from IDEs like auto-completion for example and they are not funny to implement.

> The builder patten: column.push(a).push(b) instead of Elmy column([a,b]).

As @yusdacra already mentioned you can use `Column::with_children(vec![a, b])`. Maybe we could implement the `From<Vec<Element<...>>>` trait for Column and other container like widgets to even shorten this down to:
```Rust
let foo: Column<_> = vec![a, b].into();
```

**Comment by hecrj** [comment, marked as answer]
No particular reason! We just haven't prioritized reducing the verbosity of the API.

As people pointed out, nothing is stopping you from creating your own helper functions or methods!

I think we could build a new crate with helper functions to easily build UIs on top of `iced`, additional custom themes included! Maybe we could call it `iced-ui` (analogous to [`elm-ui`](https://package.elm-lang.org/packages/mdgriffith/elm-ui/latest/))?
